src/platform/platform_worker.c

```diff
@@ -504,6 +570,11 @@ CXPLAT_THREAD_CALLBACK(CxPlatWorkerThread, Context)
             CxPlatSchedulerYield();
             State.NoWorkCount = 0;
         }
+
+        if (State.TimeNow - State.LastPoolProcessTime > DYNAMIC_POOL_PROCESSING_TIME) {
```

Comment:
Minor: I see TimeNow is updated inside CxPlatRunExecutionContexts. I feel like up here in CxPlatWorkerThread, at the beginning of the while loop, would be a more natural place for that.

For example: I'm not sure if it's a bug, but I see there's a codepath where CxPlatRunExecutionContexts fails to update TimeNow. It's not great to have to think about ECs when thinking through the correctness of this new code that uses the TimeNow variable.